src/acme/ui/AcmeUI.java

```diff
@@ -21,7 +21,10 @@ public AcmeUI() {
         setLocationRelativeTo(null);
         addWindowListener(new ShutdownListener());
 
-        reports();
+        //TODO implement shutdownlistener and remove this
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

```

Comment:
Shutdown listener is implemented, look at line 22 addWindowListener

Reply:
The problem is, the listener doesn't do anything yet. The process stays running without the default close operation here or the listener being completed.

Reply:
It is fixed on my other pull request, this will just cause merging issues later